src/java_bytecode/Makefile

```diff
@@ -36,10 +36,52 @@ INCLUDES= -I ..
 include ../config.inc
 include ../common
 
-CLEANFILES = java_bytecode$(LIBEXT)
+
+SOURCE_DIR := library/src
+BINARY_DIR := library/classes
+
+FIND := find
+
+JAVAC := javac
+JFLAGS := -sourcepath $(SOURCE_DIR) -d $(BINARY_DIR) -XDignore.symbol.file
+
+CLASSPATH := SOURCE_DIR
+
+all_javas := $(shell $(FIND) $(SOURCE_DIR) -name '*.java')
+all_classes := $(patsubst $(SOURCE_DIR)/%.java,$(BINARY_DIR)/%.class,$(all_javas))
+
+$(BINARY_DIR):
+	mkdir -p $(BINARY_DIR)
+
+.SUFFIXES: .java .class
+
+$(BINARY_DIR)/%.class : $(SOURCE_DIR)/%.java $(BINARY_DIR)
+	$(JAVAC) $(JFLAGS) $(patsubst $(BINARY_DIR)/%.class,$(SOURCE_DIR)/%.java,$@)
+
+JAR := jar
+JARFLAGS := -cf
+
+.NOTPARALLEL: core-models.jar
```

Comment:
Why?

Reply:
Java classes compile together, meaning two javac calls in this make may trigger the same .java compile for an auxiliary class. This avoids the race condition